number2/test.c

```diff
@@ -0,0 +1,49 @@
+#include "binComparison"
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+bool run_tests()
+{
+    int s = 0;
+
+    bool a[] = { 1, 0, 1 };
+    bool b[] = { 1, 1, 1 };
+    int size1 = sizeof(a) / sizeof(a[0]) int size1 = sizeof(b) / sizeof(b[0])
+
+                                                         if (!compar(a, b, size1, size2))
+    {
+        s++;
+    }
```

Comment:
```suggestion
    int size1 = sizeof(a) / sizeof(a[0]);
    int size1 = sizeof(b) / sizeof(b[0]);

    if (!compar(a, b, size1, size2)) {
        s++;
    }
```

КХМ, Вы это вообще собирали?